src/Atmos/Model/prog_prim_conversion.jl

```diff
@@ -74,11 +76,12 @@ function prognostic_to_primitive!(
     prim::Vars,
     prog::Vars,
     e_int::AbstractFloat,
+    aux::Vars,
 )
-    ts = PhaseDry(parameter_set(atmos), e_int, prog.ρ)
-    prim.ρ = prog.ρ
-    prim.u = prog.ρu ./ prog.ρ
-    prim.p = air_pressure(ts)
+    ts = new_thermo_state(atmos, prog, aux)
+    prim.ρ = air_density(ts) # Needed for recovery of energy
+    prim.u = prog.ρu ./ density(atmos, prog, aux)
```

Comment:
What are the two different densities?

Reply:
In compressible they are the same. In anelastic, air_density(ts) returns the density from the thermodynamic state constructed by (p_ref, prog.e_int), which is not prog.ρ = aux.ref_state.ρ, but the thermodynamically consistent density.

We need this density to recover the thermo state from the primitive variables.

Reply:
Maybe we should clearly document what the two different densities are, in the `Compressibility` docs. I think it's clear from the code _how_/_where_ things are computed, but the _why_ is less clear to me. Specifically, what equations are these two different equations satisfying?

Reply:
Done!